agents/matlab/matlab_agent/src/core/interactive.py

```diff
@@ -0,0 +1,302 @@
+
+    def recv_all(self) -> list[Dict[str, Any]]:
+        if not self._conn or not select([self._conn], [], [], 0)[0]:
+            return []
+
+        self._buffer += self._conn.recv(4096)
+        lines = self._buffer.split(b"\n")
+        self._buffer = lines[-1]
+        messages: list[Dict[str, Any]] = []
+        for line in lines[:-1]:
+            line = line.strip()
+            if not line:
+                continue
+            try:
+                messages.append(json.loads(line.decode()))
+            except json.JSONDecodeError as exc:  # pragma: no cover - logging only
+                logger.error("[INTERACTIVE] Invalid JSON: %s", exc)
+        return messages
+
+    def close(self) -> None:
+        if self._conn:
+            self._conn.close()
+        if self._srv:
+            self._srv.close()
+        if self.matlab_proc and self.matlab_proc.poll() is None:
+            self.matlab_proc.terminate()
+            try:
+                self.matlab_proc.wait(timeout=5)
+            except subprocess.TimeoutExpired:  # pragma: no cover - best effort
+                self.matlab_proc.kill()
+
+
+# ---------------------------------------------------------------------------
+# Controller
+# ---------------------------------------------------------------------------
+
+class MatlabInteractiveController:
+    def __init__(
+        self,
+        path: str,
+        file: str,
+        source: str,
+        broker: IMessageBroker,
+        templates: Dict[str, Any],
+        tcp_cfg: Dict[str, Any],
+        bridge_meta: str,
+        request_id: str,
+        agent_id: str = "agent",
+    ) -> None:
+        self.sim_path = Path(path).resolve()
+        self.sim_file = file
+        if not (self.sim_path / self.sim_file).exists():
+            raise FileNotFoundError(self.sim_file)
+
+        self.source = source
+        self.broker = broker
+        self.templates = templates
+        self.bridge_meta = bridge_meta
+        self.request_id = request_id
+        self.agent_id = agent_id
+
+        self.out_srv = _TcpServer(
+            tcp_cfg.get("output_host", "localhost"),
+            tcp_cfg.get("output_port", 5678),
+        )
+        self.in_srv = _TcpServer(
+            tcp_cfg.get("input_host", "localhost"),
+            tcp_cfg.get("input_port", 5679),
+        )
+
+        self.start_time: Optional[float] = None
+        self.sequence = 0
+
+    # ------------------------------------------------------------------
+    def _start_matlab(self) -> None:
+        cmd = [
+            "matlab",
+            "-batch",
+            f"addpath('{
+                self.sim_path}');cd('{
+                self.sim_path}');run('{
+                self.sim_file}');",
+        ]
+        self.out_srv.matlab_proc = subprocess.Popen(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+
+    def start(self, pm: PerformanceMonitor) -> None:
```

Comment:
(bridgeID,operationID,Simulation Type) are the most relevant fields to be used for performance monitoring.